src/FTPClientWrapperSSH.cpp

```diff
@@ -655,6 +622,8 @@ int FTPClientWrapperSSH::verify_knownhost(ssh_session session) {
 	if (rc < 0)
 		return -1;
 
+	const char *keytype = ssh_key_type_to_char(ssh_key_type(srv_pubkey));

```

Comment:
From docu:

Returns
    A string for the keytype or NULL if unknown. 

So probably we need a check for the NULL scenario, also it is not stated clearly if this happens for SSH_KEYTYPE_UNKNOWN or some casted non-enum value